15-Appendix.Rmd

```diff
@@ -0,0 +1,144 @@
+
+### Cross Language
+
+Both `dbtools` and `s3tools` rely on Python packages through the `reticulate` R
+to Python bridge. This means that `packrat` is not enough to reproducibly and
+reliably manage all of your application's dependencies.
+
+`conda` supports managing both Python and R dependencies in a single
+`environment`. It can make sure all of these libraries are compatible with each
+other.
+
+## Installing Packages
+
+If you need to find a package name you can use the [anaconda search] to find the
+name. Run `conda install PACKAGENAME` in the `Terminal` tab to install it.
+
+### R
+
+Most CRAN (around 95%) are available through `conda`, they retain the same name
+as the CRAN package name with an addition `r-` prefix. This is to avoid clashes
+with Python packages with the same name.
+
+#### Examples:
+
+##### Install a package
+
+| `install.packages` (in R-Console) | `conda install` (in Terminal) |
+| --------------------------------- | ----------------------------- |
+| `install.packages('Rcpp')`        | `conda install r-Rcpp`        |
+
+![](images/conda/conda_install_rcpp.gif)
+
+##### Install a specific version of a package
+
+| `install.packages`                                                                                              | conda install                   |
+| --------------------------------------------------------------------------------------------------------------- | ------------------------------- |
+| `require(devtools)`<br> `install_version("ggplot2", version = "2.2.1", repos = "http://cran.us.r-project.org")` | `conda install r-ggplot2=2.2.1` |
+
+![](images/conda/conda_install_specific_version.gif)
+
+### Python
+
+Python packages do not require a prefix and can simply be installed using their
+name.
+
+#### Examples
+
+##### Install a package
+
+In the terminal run: `conda install numpy`
+
+which you can now access in your R session
+
+```r
+library(reticulate)
+np <- import("numpy")
+np$arange(15)
+```
+
+![](images/conda/np_from_conda.gif)
+
+### Operating System Packages
+
+Even if you want to continue using `packrat` to manage your R packages you can
+use conda to resolve operating system dependencies like libxml2
+
+#### Examples
+
+##### Installing a package that relies on OS dependency
+
+Suppose you want to install the R package `bigIntegerAlgos` but it fails because
+it depends on a system level library called `gmp`. Switch to the terminal and
+use `conda` to install it. Then switch back to the R console and try to use
+`install.packages` again
+
+![](images/conda/conda_install_with_os_dep_v2.gif)
+
+## Platform limitations
+
+Usually when using Conda, it makes sense to have one environment per project,
+but because we are using the Open Source version of R Studio, there is only a
+single Conda environment available.
+
+This means having to be careful to make sure packages don't pollute your
+environment from another project.
```

Comment:
Is there a specific way to deal with this?

Reply:
the next section explains this but maybe that should be a subsection of this?

Reply:
I think it's fine to have it as a separate section but it might be good to signpost more explicitly.